### PR TITLE
Refactor parser parse_stmt method

### PR DESCRIPTION
**Pull request by DavidKnott**
### - What I did
I broke the `parse_stmt` method out into it's own class.

### - How I did it
1) Turned the `parse_stmt` conditional into a dictionary
2) Move `parser.py` into a parser folder
3) Create a seperate `stmt.py` file

### - How to verify it
Look through my PR to make sure my refactors increase simplicity and don't change any functionality.

### - Description for the changelog
None
### - Cute Animal Picture
image

### CODE REVIEW

**Comment by fubuloubu** [comment]
2 Failed tests in Travis. Why?

**Comment by DavidKnott** [comment]
It's because i hardcoded the timestamps in for the logging tests and version 2.0.5 has diferent starting timetamps.  My other PR fixes this.

File: viper/parser/stmt.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+            ast.AnnAssign: self.ann_assign,
+            ast.Assign: self.assign,
+            ast.If: self.parse_if,
+            ast.Call: self.call,
+            ast.Assert: self.parse_assert,
+            ast.For: self.parse_for,
+            ast.AugAssign: self.aug_assign,
+            ast.Break: self.parse_break,
+            ast.Return: self.parse_return,
+        }
```
Comment (jacqueswww) [inline viper/parser/stmt.py]: Just something small thing I would opt for readability. I like to have something like stmt_table on the class straight: 

```python

class Stmt():
   stmt_table = { ... }

File: viper/parser/stmt.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        return LLLnode.from_list(['if', parse_value_expr(self.stmt.test, self.context), parse_body(self.stmt.body, self.context)] + add_on, typ=None, pos=getpos(self.stmt))
+
+    def call(self):
+        from .parser import (
+            parse_expr,
+            pack_arguments,
+            pack_logging_data,
+            pack_logging_topics,
+            external_contract_call_stmt,
+        )
```
Comment (jacqueswww) [inline viper/parser/stmt.py]: Coul we add # TODO here so we can evaluate the inline imports later  (I think once we factor everything from parse.py it should be doable) ;)

File: viper/parser/stmt.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+                                        typ=None, pos=getpos(self.stmt))
+        elif isinstance(self.stmt.func, ast.Attribute) and isinstance(self.stmt.func.value, ast.Call):
+            return external_contract_call_stmt(self.stmt, self.context)
+        elif isinstance(self.stmt.func, ast.Attribute) and self.stmt.func.value.id == 'log':
+            if self.stmt.func.attr not in self.context.sigs['self']:
+                raise VariableDeclarationException("Event not declared yet: %s" % self.stmt.func.attr)
+            event = self.context.sigs['self'][self.stmt.func.attr]
+            topics, stored_topics, event, data = pack_logging_topics(event, self.stmt.args, self.context)
+            inargs, inargsize, inarg_start = pack_logging_data(event, data, self.context)
+            return LLLnode.from_list(['seq', inargs, stored_topics, ["log" + str(len(topics)), inarg_start, inargsize] + topics], typ=None, pos=getpos(self.stmt))
```
Comment (jacqueswww) [inline viper/parser/stmt.py]: Just for readability could we add some new lines after the return statements ;) I also had another idea, now that we have our own Stmt class we could perhaps replace all those isinstance() calls with self.stmt with `self.is_attribute` and `self.func_name` and `self.is_func` properties etc.. Of course this all to your liking, just a small style thing.

File: viper/parser/stmt.py (hunk 2 above)
Comment (DavidKnott) [reply viper/parser/stmt.py]: I agree 100% the inline imports are only temporary.

File: viper/parser/stmt.py (hunk 1 above)
Comment (DavidKnott) [reply viper/parser/stmt.py]: I like that a lot better, great suggestion!

File: viper/parser/stmt.py (hunk 3 above)
Comment (DavidKnott) [reply viper/parser/stmt.py]: Could you write a quick example for how `self.is_attribute` would work?  Would it be syntactic sugar for `isinstance`?

File: viper/parser/stmt.py (hunk 1 above)
Comment (jacqueswww) [reply viper/parser/stmt.py]: Sure, makes sense, since it's just another way to do a bunch if/elifs.

File: viper/parser/stmt.py (hunk 3 above)
Comment (jacqueswww) [reply viper/parser/stmt.py]: So the idea was just as a bunch of handy aliases:

```

class Stmt():

    @property
    def is_attribute(self):
        return isinstance(self.stmt.func, ast.Attribute)

    @property
    def is_name(self):
        return isinstance(self.stmt, ast.Name) 

    @property
    def func_id(self):
        return getattr(self.stmt.func.value, 'id', None)
```

**Pull request merged**